docs/hello_nextflow/05_hello_operators.md

```diff
@@ -0,0 +1,854 @@
+
+### 2.4. Add a call to the workflow block to run GATK_GENOMICSDB
+
+We've got a process, and we've got input channels. We just need to add the process call.
+
+```groovy title="hello-operators.nf" linenums="122"
+// Combine GVCFs into a GenomicsDB datastore
+GATK_GENOMICSDB(
+    all_gvcfs,
+    all_idxs,
+    params.cohort_name
+)
+```
+
+Ok, everything is wired up.
+
+### 2.5. Run the workflow
+
+Let's see if this works.
+
+```bash
+nextflow run hello-operators.nf -resume
+```
+
+It run fairly quickly, since we're running with `-resume`, but...
+
+```console title="Output"
+ N E X T F L O W   ~  version 24.02.0-edge
+
+ ┃ Launching `hello-operators.nf` [mad_edison] DSL2 - revision: 6aea0cfded
+
+executor >  local (1)
+[a2/56a3a8] SAMTOOLS_INDEX (1)       | 3 of 3, cached: 3 ✔
+[ce/096ac6] GATK_HAPLOTYPECALLER (1) | 3 of 3, cached: 3 ✔
+[df/994a06] GATK_GENOMICSDB          | 0 of 1
+ERROR ~ Error executing process > 'GATK_GENOMICSDB'
+
+Caused by:
+  Process `GATK_GENOMICSDB` terminated with an error exit status (1)
+
+Command executed:
+
+  gatk GenomicsDBImport         -V reads_father.bam.g.vcf reads_son.bam.g.vcf reads_mother.bam.g.vcf         -L intervals.bed         --genomicsdb-workspace-path family_trio_gdb
+```
+
+Ah. On the bright side, we see that Nextflow has picked up the `GATK_GENOMICSDB` process, and specifically called it just once.
+That suggests that the `collect()` approach worked, to a point.
+But, and it's a big one, the process call failed.
+
+When we dig into the console output above, we can see the command executed isn't correct.
+
+Can you spot the error?
+Look at this bit: `-V reads_father.bam.g.vcf reads_son.bam.g.vcf reads_mother.bam.g.vcf`
+
+We gave `gatk GenomicsDBImport` multiple GVCF files for a single `-V` argument, but the tool expects a separate `-V` argument for each GVCF file.
+
+As a reminder, this was the command we ran in the container:
+
+```bash
+gatk GenomicsDBImport \
+    -V reads_mother.g.vcf \
+    -V reads_father.g.vcf \
+    -V reads_son.g.vcf \
+    -L /data/ref/intervals.bed \
+    --genomicsdb-workspace-path family_trio_gdb
+```
+
+So that means we need to somehow transform our bundle of GVCF files into a properly formatted command string.
+
+### 2.6. Construct a command line with a separate `-V` argument for each input GVCF
+
+This is where Nextflow being based on Groovy comes in handy, because it's going to allow us to use some fairly straightforward string manipulations to construct the necessary command string.
+
+Specifically, using this syntax: `all_gvcfs.collect { "-V ${it}" }.join(' ')`
+
+Once again, let's break it down into its components.
+
+1. First, we take the contents of the `all_gvcfs` input channel and apply `.collect()` on it (just like earlier).
+2. That allows us to pass each individual GVCF file path in the bundle to the **closure**, `{ "-V ${it}" }`, where `${it}` refers to that GVCF, and compose a new string that appends `-V ` to the file path.
```

Comment:
The use of implict `{ "-V ${it}" }` style closures is going to be deprecated in favor of explicit closures e.g. `{ $it -> "-V ${it}" }`. It's crunch time, so maybe we don't make this change but we should avoid that syntax moving forwards. The implicit syntax is more confusing to newbies anyways.

Reply:
I can take 5 mins to make this change, it's a good call. Better explicit than implicit, I agree.

Reply:
Hmmm is there a trick to making that explicit syntax work? it's not working for me

Reply:
Phil pointed out that leading dollar sign shouldn't be there, and also (happy surprise) I can use any name instead of `it`, which means I can do `{ gvcf -> "-V ${gvcf}" }`